src/main/java/net/snowflake/client/core/SFSession.java

```diff
@@ -438,6 +438,21 @@ public void addSFSessionProperty(String propertyName, Object propertyValue) thro
           }
           break;
 
+        case MAX_JSON_STRING_LENGTH:
+          if (propertyValue != null) {
+            int maxJsonStringLength = (Integer) propertyValue;
+
+            // Update the max string length used in ObjectMapperFactory
+            ObjectMapperFactory.setMaxJsonStringLength(maxJsonStringLength);
+
+            // Update the max string length of the instantiated mapper in this class
+            mapper
```

Comment:
don't we need to reassign the mapper? let's add at least one integration test

Reply:
It does the same thing but to make it clear, I'm now reassigning the mapper.